File: pact-jvm-consumer-junit/src/main/java/au/com/dius/pact/consumer/Pact.java
package au.com.dius.pact.consumer;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * describes the interactions between a provider and a consumer.
 * The annotated method has to be of following signature:
 *
 * public PactFragment providerDef1(PactDslWithState builder) {...}
 *
 * @author pmucha
 *
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface Pact {
    
    /**
     * name of the provider
     */
    String provider();

    /**
     * name of the consumer
     */
    String consumer();

    /**
     * name of the state, the provider has to be in
     * @deprecated Provider state should be defined on the interactions
     */
    @Deprecated
    String state() default "";
}

File: core/model/src/main/kotlin/au/com/dius/pact/core/model/annotations/Pact.kt
package au.com.dius.pact.core.model.annotations

/**
 * describes the interactions between a provider and a consumer used in JUnit tests.
 * The annotated method has to be one of following signatures:
 *
 * For legacy DSL classes and request/response interactions:
 * public RequestResponsePact providerDef1(PactDslWithProvider builder) {...}
 *
 * For message interactions:
 * public MessagePact providerDef1(MessagePactBuilder builder)
 *
 * For V4 DSL classes and any interaction type:
 * public V4Pact providerDef1(PactBuilder builder) {...}
 *
 * @author pmucha
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class Pact(
  /**
   * name of the provider
   */
  val provider: String = "",

  /**
   * name of the consumer
   */
  val consumer: String
)
